util/src/main/java/io/kubernetes/client/Logs.java

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client;
+
+import io.kubernetes.client.Configuration;
+import io.kubernetes.client.apis.CoreV1Api;
+import io.kubernetes.client.models.V1Pod;
+import io.kubernetes.client.util.WebSockets;
+import io.kubernetes.client.util.WebSocketStreamHandler;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.io.Reader;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.squareup.okhttp.Call;
+import com.squareup.okhttp.Response;
+
+/**
+ * Utility class offering streaming access to Pod logs.
+ */
+public class Logs {
+    private ApiClient apiClient;
+    private CoreV1Api coreClient;
+
+    /**
+     * Simple Logs API constructor, uses default configuration
+     */
+    public Logs() {
+        this(Configuration.getDefaultApiClient());
+    }
+
+    /**
+     * Logs API Constructor
+     * @param apiClient The api client to use.
+     */
+    public Logs(ApiClient apiClient) {
+        this.apiClient = apiClient;
+        this.coreClient = new CoreV1Api(apiClient);
+    }
+
+    /**
+     * Get the API client for these Logs operations.
+     * @return The API client that will be used.
+     */
+    public ApiClient getApiClient() {
+        return apiClient;
+    }
+
+    /**
+     * Set the API client for subsequent Logs operations.
+     * @param apiClient The new API client to use.
+     */
+    public void setApiClient(ApiClient apiClient) {
+        this.apiClient = apiClient;
+    }
+
+    /**
+     * Stream logs from a running container.
+     * Important note. You must close this stream or else you can leak connections.
+     * @param pod The Pod to get logs for, uses the first container in the pod for logs
+     * @return A UTF-8 encoded InputStream
+     */
+    public InputStream streamNamespacedPodLog(V1Pod pod) throws ApiException, IOException {
```

Comment:
should we just call this `streamPodLog`?

Reply:
I think this is done for consistency, all the generated api methods seem to follow `<verb>Namespaced<resource>` for this kind of stuff

Reply:
Yeah, this was for consistency...